mempool/clist_mempool.go

```diff
@@ -263,11 +263,12 @@ func (mem *CListMempool) CheckTx(tx types.Tx, sender p2p.ID) (*abcicli.ReqRes, e
 			// Note it's possible a tx is still in the cache but no longer in the mempool
 			// (eg. after committing a block, txs are removed from mempool but not cache),
 			// so we only record the sender for txs still in the mempool.
-			if elem, ok := mem.getCElement(tx.Key()); ok {
+			txKey := tx.Key()
+			if elem, ok := mem.getCElement(txKey); ok {
 				memTx := elem.Value.(*mempoolTx)
 				if found := memTx.addSender(sender); found {
 					// It should not be possible to receive twice a tx from the same sender.
-					mem.logger.Error("tx already received from peer", "tx", tx.Hash(), "sender", sender)
+					mem.logger.Error("tx already received from peer", "tx", log.NewLazySprintf("%X", txKey), "sender", sender)
```

Comment:
Thanks for the update. In the logs we should use `Hash` because it is displayed different than `Key`, and the key is internal while the hash is for the users, even when they are equivalent.